patches/tModLoader/Terraria/ModLoader/ModWaterStyle.cs

```diff
@@ -21,6 +21,9 @@ public abstract class ModWaterStyle : ModTexturedType
 		}
 
 		public sealed override void SetupContent() {
+			if (Main.dedServ)
+				return; // LiquidRenderer.Instance is null for server && Loader is client sided.
+
```

Comment:
Yes and no.
ResizeArrays is fixed; SetupContent itself is not since it has nothing to do with loaders and is auto-invoked.

I've changed it to disable autoload of water styles on the server instead as that is more elegant